language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/MatchedExpressionNodeResolver.java

```diff
@@ -46,6 +49,26 @@ public MatchedExpressionNodeResolver(Node matchedNode) {
         this.matchedNode = matchedNode;
     }
 
+    /**
+     * Given the node, this method returns the optional expression in which the provided node is located.
+     *
+     * @param node Node
+     * @return Optional enclosing expression node
+     */
+    public Optional<ExpressionNode> findExpression(Node node) {
+        if (node == null) {
+            return Optional.empty();
+        }
+
+        Optional<ExpressionNode> exprNode = node.apply(this);
+        // Due to the way apply() method is implemented in some cases, this can return null
+        if (exprNode == null) {
+            return Optional.empty();
+        }
+        
+        return exprNode;
```

Comment:
```suggestion
        return exprNode.isPresent()? exprNode : Optional.empty();
```

Is this further simplification possible?

Reply:
This might generate NPE when exprNode is null

Reply:
Can use `return exprNode == null?  Optional.empty() : exprNode;` instead?

Reply:
+1

Reply:
+1. Done!